xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -193,6 +221,7 @@ func (b *cdsBalancer) run() {
 					}
 					b.clusterToWatch = update.clusterName
 				}
+				b.ccw.setClient(b.client)
```

Comment:
Sorry, I don't have much background here..why is the xds client being swapped out over time?